### ISSUE TITLE
Problem with fetching data with Graphql API from Craft CMS v4 to Gatsby v4 page

### ISSUE THREAD

**Issue by pleodigital** [craftcms/gatsby-source-craft#108]
### Description
I have website build in Gatsby v3 which sources data from Craft CMS v3 in headless mode with graphql api. I updated both Gatsby and Craft to v4 and now I can't source data. On running gatsby develop and gatsby build I'm getting whole list of errors with all data typenames : 
ERROR #11321  PLUGIN

"gatsby-source-craft" threw an error while running the sourceNodes lifecycle:

Failed to execute query LIST_banner_module_Entry.

 Which doesn't make any sense, I don't have any data typename starting with **"LIST_"**. 

I assume my config in Craft and it gatsby is ok because it works fine before update. 
**Craft CMS config:**
general.php file:  'enableGql' => true, 'headlessMode' => true,
composer.json: "craftcms/gatsby-helper": "2.0.1",
**Gatsby config:**
`gatsby-source-craft`,
				options: {
					craftGqlToken: process.env.CRAFTGQL_TOKEN,
					craftGqlUrl: process.env.CRAFTGQL_URL,
					enabledSites: [`default`],
					retryOptions: { retries: 15 },
				},

### Steps to reproduce

1. Upgrade Craft CMS from v3 to v4 (and all needed plugins)
2. Upgrade Gatsby from v3 to v4 (and all plugins)
3. Try to fetch data with graphql api.

### Additional info

- Craft version: 4.4.0
- Gatsby: 4.25.5
- PHP version: 8.1.6
- Database driver & version: MariaDB 10.4.24
- Craft Plugins & versions:
        "born05/craft-twofactorauthentication": "^3.0.1",
        "carlcs/craft-redactorcustomstyles": "^4.0.3",
        "craftcms/aws-s3": "^2.0.1",
        "craftcms/cms": "^4.4.0",
        "craftcms/contact-form": "^3.0.0",
        "craftcms/gatsby-helper": "2.0.0",
        "craftcms/redactor": "^3.0.3",
        "ether/logs": "^4.0.0",
        "ether/simplemap": "^4.0.3",
        "league/csv": "^9.8",
        "openpayu/openpayu": "^2.3",
        "paypal/paypal-checkout-sdk": "^1.0",
        "pleodigital/freshmail": "v1.0.13",
        "sebastianlenz/craft-chunked-uploads": "^2.0.1",
        "vlucas/phpdotenv": "^3.4.0",
        "wrav/oembed": "^2.3.0"
- Gatsby plugins:
 "@sentry/gatsby": "^6.14.1",
    "axios": "^0.19.2",
    "fb-react-sdk": "^1.0.4",
    "freshmail-js": "^1.0.1",
    "gatsby": "4.25.5",
    "gatsby-cli": "^4.20.0",
    "gatsby-image": "^2.4.21",
    "gatsby-plugin-facebook-pixel": "^1.0.5",
    "gatsby-plugin-google-gtag": "^4.0.0",
    "gatsby-plugin-google-tagmanager": "^4.0.0",
    "gatsby-plugin-load-script": "^1.0.6",
    "gatsby-plugin-manifest": "^4.0.0",
    "gatsby-plugin-react-helmet": "^5.0.0",
    "gatsby-plugin-robots-txt": "1.6.8",
    "gatsby-plugin-sharp": "^4.0.0",
    "gatsby-plugin-sitemap": "^5.0.0",
    "gatsby-plugin-styled-components": "^3.3.14",
    "gatsby-plugin-web-font-loader": "^1.0.4",
    "gatsby-source-craft": "3.0.0",
    "gatsby-source-filesystem": "^4.0.0",
    "gatsby-source-twitter": "^4.0.1",
    "gatsby-transformer-sharp": "^4.0.0",
    "google-map-react": "^1.1.7",
    "isotope-layout": "^3.0.6",
    "js-cookie": "^3.0.1",
    "lottie-web": "^5.8.1",
    "prop-types": "^15.8.1",
    "public-ip": "^4.0.2",
    "react": "^16.13.1",
    "react-cookie-consent": "^2.5.0",
    "react-device-detect": "^1.14.0",
    "react-dom": "^16.13.1",
    "react-dropdown": "^1.9.0",
    "react-helmet": "^5.2.1",
    "react-load-script": "0.0.6",
    "react-moment": "^0.9.7",
    "react-player": "^1.15.3",
    "react-pose": "^4.0.10",
    "react-range": "^1.8.12",
    "react-redux": "^7.2.6",
    "react-scrollspy": "^3.4.3",
    "react-share": "^4.3.1",
    "react-slick": "^0.25.2",
    "redux": "^4.1.2",
    "sha1": "^1.1.1",
    "sha256": "^0.2.0",
    "sharp": "^0.23.4",
    "slick-carousel": "^1.8.1",
    "styled-components": "^4.4.1",
    "twitter": "^1.7.1"

I can't find a way to fetch my data, do you have any ideas? What's interesting, when I'm trying to fetch data to my updated Gatsby v4 page from not updated Craft 3 everything works fine.

**Comment by brianjhanson** [comment]
The `LIST_` prefix is a Gastby convention. In the initial sourcing, it creates it's own queries (aided by our source plugin) in order to get all the content from Craft. 

Are there any additional logs that might point toward the problem? Could you clear the gatsby cache and give it another try if you haven't already?

**Comment by pleodigital** [comment]
My logs: [web-2023-03-20.log](https://github.com/craftcms/gatsby-source-craft/files/11015855/web-2023-03-20.log) . I'm clearing gatsby cache before every 'develop' or 'build' command.

**Comment by brianjhanson** [comment]
Looks like you have a call to `getSourceUid()` somewhere (potentially by having `sourceUid` within your graphql query). That method was deprecated in 4.0 and has been replaced by `getCanonicalUid()`. If you have any `sourceUid` properties in your query, can you replace them with `canonicalUid` and see if that solves it? 

You'll probably want to do a pass through [the upgrade guide](https://craftcms.com/docs/4.x/upgrade.html#breaking-changes-and-deprecations) and make sure you don't have any other deprecated methods hanging around.

**Comment by pleodigital** [comment]
I don't have any getSourceUid() or sourceUid in my code. At this moment I have blank Gatsby v4 project with no graphql queries at all, I just want to run gatsby develop without error to run graphiQL. I don't think gatsby setup can be wrong, because it works with my project with craft cms v3 (version before updating to v4).

**Comment by brianjhanson** [comment]
The main error in the logs is

```
2023-03-20 08:54:40 [web.ERROR] [craft\errors\DeprecationException] craft\errors\DeprecationException: Elements’ `getSourceId()` method has been deprecated. Use `getCanonicalId()` instead. in C:\xampp\htdocs\craft-polska-misja-medyczna\vendor\craftcms\cms\src\gql\base\ObjectType.php:72
Stack trace:
```

Did you update the Gatsby helper plugin along with Craft and the Gatsby source plugin?

**Comment by pleodigital** [comment]
Yes, I have Gatsby Helper v2.0.1 installed. I've also tried with 2.0.0 but with no result. Craft version is 4.4.3, gatsby-source-craft v.3.0.0. Everything has the newest possible version, maybe there's some combination of previous versions to make it work?

**Comment by brianjhanson** [comment]
Unfortunately, I'm not able to replicate this on a fresh install. Could you package up a db dump, the gatsby files, your `composer.json` and `composer.lock` and send it to `support@craftcms.com`?

**Comment by pleodigital** [comment]
I sent the files you requested, but accidentally used a private email instead of a company email, so please ignore first email and use files from second one (this from info@pleodigital.com). In first email I sent wrong composer files. Sorry for the trouble.

**Comment by brianjhanson** [comment]
All good! I got them, I'll take a look and let you know what I find. Thanks!

**Comment by pleodigital** [comment]
The solution to the problem turned out to be simple, but not obvious to come up with. I needed to create my GraphQL schema and API token once again. Looks like it was problem with recognition of schema created with Craft 3. Or maybe it's just a coincidence, either way it works :)
